cdi/instance-qualifiers/src/test/java/org/javaee7/cdi/instance/GreetingTest.java

```diff
@@ -0,0 +1,68 @@
+package org.javaee7.cdi.instance;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Inject;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.*;
+
+/**
+ * @author Radim Hanus
+ */
+@RunWith(Arquillian.class)
+public class GreetingTest {
+	@Deployment
+	public static Archive<?> deploy() {
+		return ShrinkWrap.create(JavaArchive.class)
+				.addClasses(Greeting.class, SimpleGreeting.class, FormalGreeting.class, Business.class, Personal.class)
+				.addAsManifestResource("beans.xml");
+	}
+
+	/**
+	 * Container will assume built-in @Default qualifier here as well as for beans that don't declare a qualifier.
+	 */
+	@Inject
+	private Instance<Greeting> defaultInstance;
+
+	/**
+	 * Qualifier @Personal is not qualifying any bean.
+	 */
+	@Inject @Personal
+	private Instance<Greeting> personalInstance;
+
+	/**
+	 * Built-in qualifier @Any is assumed on each bean regardless other qualifiers specified.
+	 */
+	@Inject @Any
+	private Instance<Greeting> anyInstance;
+
+	@Test
+	public void test() throws Exception {
```

Comment:
I think it would be better to split to smaller, descriptive tests. But I guess I'm too late to the party, as it is merged anyway.